### PR TITLE
[tf] New authz server component

### PR DESCRIPTION
**Pull request by nmittler**
Moving the authz server creation to a new test component that can be easily reused.

This is supporting the work on https://github.com/istio/istio/pull/37914.

**Please provide a description of this PR:**

### CODE REVIEW

File: pkg/test/framework/components/authz/kube.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+		},
+	}
+	server.id = ctx.TrackResource(server)
+
+	// Deploy the authz server.
+	if err = server.deploy(ctx); err != nil {
+		return
+	}
+
+	// Patch MeshConfig to intall the providers.
```
Comment (yangminzhu) [inline pkg/test/framework/components/authz/kube.go:122]: s/intall/install

File: pkg/test/framework/components/authz/provider.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+// Provider for authz requests.
+type Provider interface {
+	Name() string
+
+	// API used by this provider.
+	API() API
+
+	// IsProtocolSupported returns true if the given request protocol is supported by this provider.
+	IsProtocolSupported(protocol.Instance) bool
```
Comment (yangminzhu) [inline pkg/test/framework/components/authz/provider.go:45]: Are these methods actually used in the test? I found only the `Name()` and `API()` is used. If they are needed in a follow-up PR, it's better to add in the follow-up PR together for easier review.

**Review by howardjohn — APPROVED** [review]

File: pkg/test/framework/components/authz/server.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	t.Helper()
+	s, err := NewLocal(t, ns)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return s
+}
+
+// Setup is a utility function for configuring a global authz Server.
+func Setup(server *Server, ns func() namespace.Instance) resource.SetupFn {
```
Comment (yangminzhu) [inline pkg/test/framework/components/authz/server.go:64]: The `Setup()` and `SetupLocal()` is also not used, probably delete them in this PR and add later when actually used.

**Review by yangminzhu — COMMENTED** [review]
looks good, just some small comments.

**Comment by nmittler** [comment]
@howardjohn @ericvn @GregHanson FYI the build seems broken currently. I looked briefly at history but nothing stands out. Is anyone looking into this?

**Pull request merged**